src/query/sql/src/planner/semantic/type_check.rs

```diff
@@ -928,6 +941,13 @@ impl<'a> TypeChecker<'a> {
             arg_types.push(arg_type);
         }
 
+        // rewrite substr('xx', 0, xx) -> substr('xx', 1, xx)
+        if self.ctx.get_settings().get_sql_dialect().unwrap() == Dialect::Hive
```

Comment:
new method like  `fn index_zero_literal_as_one(&self) -> bool`

Reply:
add this method in Dialect?